### ISSUE TITLE
ProjectType project gives "VSSDK1031: Extension error" on build after uninstalling the extension in the experimental instance

### ISSUE THREAD

**Issue by davidkron** [Microsoft/VSProjectSystem#74]
To reproduce:

1. Create a new projecttype project
2. Ensure the solution builds
3. Launch the ProjectType project in the experimental instance (F5)
4. Go to "Tools" -> "Extensions and updates" on the experimental instance
5. Find the extension matching the created projecttype and uninstall it
6. Close the experimental instance
7. Right click on the solution and select "rebuild"

```
1>------ Rebuild All started: Project: ProjectType3.ProjectTemplate, Configuration: Debug Any CPU ------
1>  ProjectType3.ProjectTemplate -> D:\dev\csharp\DevArch\ProjectType3\ProjectType3\ProjectType3.ProjectTemplate\bin\Debug\ProjectTemplates\ProjectType3\1033\ProjectType3.zip
1>  ProjectType3.ProjectTemplate -> 
2>------ Rebuild All started: Project: ProjectType3.ProjectType, Configuration: Debug Any CPU ------
2>  ProjectType3.ProjectType -> 
2>  ProjectType3.ProjectType -> 
2>  ProjectType3.ProjectType -> D:\dev\csharp\DevArch\ProjectType3\ProjectType3\ProjectType3.ProjectType\bin\Debug\ProjectType3.dll
2>  Creating intermediate PkgDef file.
2>  Creating VSIX Container...
2>  ProjectType3.ProjectType -> D:\dev\csharp\DevArch\ProjectType3\ProjectType3\ProjectType3.ProjectType\bin\Debug\ProjectType3.vsix
2>C:\Program Files (x86)\MSBuild\Microsoft\VisualStudio\v14.0\VSSDK\Microsoft.VsSDK.targets(653,5): error VSSDK1031: Extension 'ProjectType3.ProjectType..46f10d4a-3105-49a3-8cec-0d401288da4f' could not be found. Please make sure the extension has been installed.
========== Rebuild All: 1 succeeded, 1 failed, 0 skipped ==========
```

This also occurs if the two projects are wrapped in another vsix container.

**Comment by davidkron** [comment]
Changing the version number in the *source.extension.vsixmanifest* file is an acceptable workaround that doesn't seem to problematic. 

I vote to close this for now, with this information kept for reference.

**Comment by rprouse** [comment]
Another workaround is to delete your experimental instance. For example, for Visual Studio 2015, delete the folder

C:\Users\{user}\AppData\Local\Microsoft\VisualStudio\14.0Exp

**Comment by AArnott** [comment]
I *believe* another workaround is to simply relaunch 'devenv /rootsuffix exp' yourself, and close. After that, I think it works.

**Comment by Gmaniacuk1** [comment]
You can always reset the experimental instance, although up-ing the version number worked for me.

To reset the experimental instance:
http://www.codeproject.com/Tips/832362/Resetting-the-Visual-Studio-Experimental-Instance
https://msdn.microsoft.com/en-us/library/dd875761.aspx

**Comment by dmihailescu** [comment]
under the Project properties go to the VSIX tab.
UNcheck 'Deploy VSIX content to experimental instance for debugging' and build.
If you needed to debug, check it back and build again.

**Comment by yannduran** [comment]
Hmm, here we are in 2023 now! I found this issue because I had the very same error and this was the top of the search results. I'm glad this issue was here to help me almost 8 years later.

The weird thing is I've uninstalled the extension that I've been working on many times from the Experimental instance (when my current version number is less than, or equal to the installed version), and I've  never encountered this error before. But it occurred today, just as explained.

@AArnott's suggestion of course did the trick, but I used Mads Kristensen's **_Clear MEF Component Cache_** extension, but I did it in the instance of VS that I'm developing with, and didn't think to do it in the Experimental instance, as I've never needed to do that before.

Thanks!
